src/mathematics/glyphinfo.rs

```diff
@@ -0,0 +1,148 @@
+use truetype::{Result, Tape, Walue};
+use std::io::SeekFrom;
+
+use layout::Coverage;
+use super::Quantity;
+
+table! {
+    @position
+    #[doc = "A table of positioning information defined on a per-glyph basis."]
+    pub Glyphs { // MathGlyphInfo
+        italics_corrections_offset     (u16), // MathItalicsCorrectionInfo
+        top_accent_attachments_offset  (u16), // MathTopAccentAttachment
+        extended_shape_coverage_offset (u16), // ExtendedShapeCoverage
+        kernings_offset                (u16), // MathKernInfo
+
+        corrections (Corrections) |this, tape, position| {
+            jump_take!(tape, position, this.italics_corrections_offset)        
+        },
+
+        attachments (Attachments) |this, tape, position| {
+            jump_take!(tape, position, this.top_accent_attachments_offset)
+        },
+
+        extended_shape_coverage (Option<Coverage>) |this, tape, position| {
+            jump_take_maybe!(tape, position, this.extended_shape_coverage_offset)
+        },
+
+        // According to the specification, kernings are not optional.  However,
+        // some popular OpenType fonts do not support kernings.
+        kernings (Option<Kernings>) |this, tape, position| {
```

Comment:
I decided to remove the `Option`s from `corrections` and `attachments`, but kept it on `kernings` with a note about deviating from the specification.  I should test more OpenType math fonts -- there is only a half dozen or so.